cantools/database/diagnostics/data.py

```diff
@@ -79,30 +73,6 @@ def byte_order(self):
     def byte_order(self, value):
         self._byte_order = value
```

Comment:
IMO it would be good to handle all "simple" properties uniformly, not just the performance-critical ones...

Reply:
Yes i agree. I didnt want to change more than necessary to make the review easier. Once we agree on the change i will update the other properties.

Reply:
I converted the properties of Signal and Data. Tell me if i missed something

Reply:
looks good. It seems like cpython does not pick it up for the class' docstring, but sphinx does. While not optimal, I think this is an acceptable trade-off...